app/react/Pages/components/PageViewer.js

```diff
@@ -12,12 +12,14 @@ import Script from './Script';
 
 const { Context } = MDComponents;
 
-export class PageViewer extends Component {
+class PageViewer extends Component {
   render() {
     const { page, itemLists, datasets } = this.props;
     const lists = itemLists.toJS();
     const originalText = page.getIn(['metadata', 'content']) || '';
-    const scriptCode = page.getIn(['metadata', 'script']) || '';
+    let scriptCode = page.getIn(['metadata', 'script']) || '';
+
+    scriptCode = `var datasets=${JSON.stringify(datasets)};\r\n${scriptCode}`;
```

Comment:
was this really part of the issue ? or absolutely needed ? i think we should discuss this first ? we are now coupling markdown datasets with the script that was thought to be self contained ?

Reply:
This was requested by @txau precisely as a intermediate step into "dynamic" access to properties.  I actually think it is monstrously useful and really don't know why we didn't do this sooner.  

If you don't think it a good feature, yes, lets please discuss this.

Reply:
i understand the utility, but some of this datasets are huge, and we already have them loaded like this on some scenarios, for instance when you do SSR they are on ```window.__reduxData__.page.datasets``` we will duplicate what already is really big in this scenario.

maybe an alternative is to give the script access to the store itself, that has all that data loaded in memory already which give the script access not only to datasets, but also to everything else ? i guess its potentially a security issue ? but come on, this free JS script its already that.

Reply:
Moving this development to a new issue.